unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsHUD/SettingsGeneralView.cs

```diff
@@ -162,6 +167,21 @@ void Awake()
                 tempGeneralSetting.voiceChatAllow = (DCL.SettingsData.GeneralSettings.VoiceChatAllow)value;
                 isDirty = true;
             });
+
+            autosettingsToggle.onValueChanged.AddListener(value =>
+            {
+                advancedCanvasGroup.interactable = !value;
+                tempGeneralSetting.autoqualityOn = value;
+                advancedBlocker.SetActive(value);
+                if (value)
+                {
+                    QualitySettings.BaseResolution currentBaseResolution = tempQualitySetting.baseResolution;
+                    tempQualitySetting = Settings.i.lastValidAutoqualitySet;
+                    tempQualitySetting.baseResolution = currentBaseResolution;
+                    isDirty = true;
+                }
```

Comment:
In order to improve the code readability, I think it would be fine to extract this piece of code to a private function (something like `SetAutoSettingsActive(bool active)`). What do you think?